AppHandling/Run-AlPipeline.ps1

```diff
@@ -875,9 +875,12 @@ if ($updateLaunchJson) {
 
 if ($useCompilerFolder -or $filesOnly -or !$useDevEndpoint) {
     $packagesFolder = CheckRelativePath -baseFolder $baseFolder -sharedFolder $sharedFolder -path $packagesFolder -name "packagesFolder"
-    if (!(Test-Path $packagesFolder)) {
-        New-Item $packagesFolder -ItemType Directory | Out-Null
+    if (!($bcContainerHelperConfig.doNotRemovePackagesFolderIfExists)) {
+        if (Test-Path $packagesFolder) {
+            Remove-Item $packagesFolder -Recurse -Force
+        }
     }
+    New-Item $packagesFolder -ItemType Directory -Force | Out-Null 

```

Comment:
If the folder already exists - won't this fail?

Reply:
I think this needs to be something like::
```
    if (Test-Path $packagesFolder) {
        if (!($bcContainerHelperConfig.doNotRemovePackagesFolderIfExists)) {
            Remove-Item $packagesFolder -Recurse -Force
            New-Item $packagesFolder -ItemType Directory -Force | Out-Null 
        }
    }
    else {
        New-Item $packagesFolder -ItemType Directory -Force | Out-Null 
    }
```

Reply:
No according to the documentation the force will not fail for directories. The behavavior is that it will create if the folder exists and skip if it not exists

See documentation of New-Item here
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-item?view=powershell-7.5#example-8-use-the-force-parameter-to-attempt-to-recreate-folders

Reply:
Got it - and that also works on 5.1 - thanks